Description: Fix build on arm64
Author: Emanuele Rocca <ema@debian.org>
Last-Update: 2022-09-12

Index: uhub-0.4.1/src/system.h
===================================================================
--- uhub-0.4.1.orig/src/system.h
+++ uhub-0.4.1/src/system.h
@@ -202,6 +202,10 @@
 #define CPUINFO "ARM"
 #endif
 
+#if defined(__aarch64__)
+#define CPUINFO "AArch64"
+#endif
+
 #if defined(__i386__) || defined(__i386) || defined(i386) || defined(_M_IX86) || defined(__X86__) || defined(_X86_) || defined(__I86__) || defined(__INTEL__) || defined(__THW_INTEL__)
 #define CPUINFO "i386"
 #endif
